tests/pipeline/pipeline-tests.groovy

```diff
@@ -1,19 +1,35 @@
-// Entering
-
-@Library("dotnet-ci") _
+// To figure out what we import, we need to know whether this is a PR or not.
+// If a PR, and GitHub, we grab the source branch, which must live in the same repo.
+// Note that we can't use the library isPR for this.
+boolean isPRTest = false
+String repository = env["ghprbGhRepository"]
+String libraryImportBranch
+if (repository != null && repository != "") {
+    echo "This is a GitHub PR test"
+    // Check that the PR source branch came from the correct location (otherwise the tests are expected to fail)
+    if (repository != 'dotnet/dotnet-ci') {
```

Comment:
The pipeline itself is pulling the library dependency from the PR source branch.  A few lines below you can see that it loads the pipeline.